OpenMarket/OpenMarket/Network/Completionable.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  Completionable.swift
+//  OpenMarket
+//
+//  Created by 이정민 on 2022/11/15.
+//
+
+protocol Completionable {}
+
+typealias StatusCode = Int
+
+extension StatusCode: Completionable {}
+extension ProductsList: Completionable {}
+extension Product: Completionable {}
```

Comment:
extension으로 프로토콜을 채택하신 이유가 있나요?
이 방법보다 각 타입을 정의할때 명시해주는건 어떤가요?
```suggestion
struct ProductList: Coadble, Completionable { 
   ... 
}
struct Product: Coadble, Completionable { 
   ... 
}
```
}

Reply:
원래는 각 타입에 명시를 해주었는데 만약 타입이 엄청 많아진다면 해당 프로토콜을 채택한 타입이 무엇인지 찾기 어렵지 않을까 생각했습니다. POP같은 방법으로 기능을 구현해야 한다면 타입에 명시를 해주는게 좋다고 생각하지만 Completionable 을 채택만하고 있는 타입들을 한눈에 보기위해서 Completionable에서 달아주었습니다.

이런 상황에서도 타입에 명시를 해주는 것이 좋았을까요?